repair/row_level.cc

```diff
@@ -1325,6 +1327,43 @@ class repair_meta {
         // Clear gently to avoid stalls
         utils::clear_gently(row_diff).get();
     }
+public:
+    const std::unique_ptr<const locator::token_metadata>& get_tm_for_small_table_optimization_check(const locator::token_metadata* tm) {
+        if (_small_table_optimization_tm_calculated) {
+            return _small_table_optimization_tm;
+        }
+        if (!tm) {
+            return _small_table_optimization_tm;
```

Comment:
What guarantees that when `!tm`, `_small_table_optimization_tm` is already calculated?

Reply:
if tm is nullptr, it is ok to return a empty _small_table_optimization_tm because there is no tm provided.

Reply:
I changed it to throw in casee a !tm is passed.